common/lib/xmodule/xmodule/modulestore/xml_exporter.py

```diff
@@ -153,20 +153,15 @@ def export(self):
         Perform the export given the parameters handed to this class at init.
         """
         with self.modulestore.bulk_operations(self.courselike_key):
-            # depth = None: Traverses down the entire course structure.
-            # lazy = False: Loads and caches all block definitions during traversal for fast access later
-            #               -and- to eliminate many round-trips to read individual definitions.
-            # Why these parameters? Because a course export needs to access all the course block information
-            # eventually. Accessing it all now at the beginning increases performance of the export.
-            fsm = OSFS(self.root_dir)
-            courselike = self.get_courselike()
-            export_fs = courselike.runtime.export_fs = fsm.makeopendir(self.target_dir)
-            root_courselike_dir = self.root_dir + '/' + self.target_dir
 
+            fsm = OSFS(self.root_dir)
             root = lxml.etree.Element('unknown')  # pylint: disable=no-member
 
             # export only the published content
             with self.modulestore.branch_setting(ModuleStoreEnum.Branch.published_only, self.courselike_key):
+                courselike = self.get_courselike()
```

Comment:
The above was the key bug here. The course was being retrieved outside the `branch_setting` context manager that restricted course items to published items only.